src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Elastic.Apm.Api;
+
+namespace Elastic.Apm.Metrics.MetricsProvider
+{
+	internal class SystemTotalCpuProvider : IMetricsProvider
+	{
+		private const string SystemCpuTotalPct = "system.cpu.total.norm.pct";
+
+		public SystemTotalCpuProvider()
+		{
+			if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) return;
+
+			var procStatValues = ReadProcStat();
+			if (!procStatValues.success) return;
+
+			_prevIdleTime = procStatValues.idle;
+			_prevTotalTime = procStatValues.total;
+		}
+
+		private double _prevIdleTime;
+		private double _prevTotalTime;
+		private PerformanceCounter _processorTimePerfCounter;
+
+		public int ConsecutiveNumberOfFailedReads { get; set; }
+		public string DbgName => "total system CPU time";
+
+		private static (bool success, double idle, int total) ReadProcStat()
+		{
+			if (!File.Exists("/proc/stat")) return (false, 0, 0);
+
+			using (var sr = new StreamReader("/proc/stat"))
+			{
+				var firstLine = sr.ReadLine();
+				if (firstLine == null || !firstLine.ToLower().StartsWith("cpu")) return (false, 0 ,0);
+
+				var values = firstLine.Substring(5, firstLine.Length - 5).Split(' ');
```

Comment:
I wouldn't rely on number of spaces following `"cpu"` (in this case 2 spaces)

`cpu  74608   2520   24433   1117073   6176   4054  0        0      0      0`

I would just take the rest of the line after `cpu` and trim it.

Reply:
Done, also added some tests covering this.